### PR TITLE
Updates from shipping apps

### PR DESCRIPTION
**Pull request by peculiar**
This PR will be updated once the PRs against AppCore have been accepted.

### CODE REVIEW

File: Asthma.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -955,7 +975,7 @@
 					"-Wextra",
 				);
 				PRODUCT_NAME = "$(TARGET_NAME)";
-				PROVISIONING_PROFILE = "";
+				PROVISIONING_PROFILE = "30227e0f-74e5-4aec-9e53-802317c13f85";
```
Comment (YuanZhu-apple) [inline Asthma.xcodeproj/project.pbxproj]: PROVISIONING_PROFILE was added back through merging. Please remove.

File: Asthma/Startup/APHAppDelegate.m [hunk 2]
```diff
@@ -210,8 +307,7 @@ - (void) setUpAppAppearance
                                                             }];
     
     [[UIView appearance] setTintColor:[UIColor appPrimaryColor]];
-    
-    self.dataSubstrate.parameters.bypassServer = YES;
+    self.dataSubstrate.parameters.hideExampleConsent = NO;
 }
```
Comment (YuanZhu-apple) [inline Asthma/Startup/APHAppDelegate.m:256]: Has "bypassServer" been done somewhere else?

File: Asthma/TabBarControllers/Dashboard/APHAirQualityDataModel.m [hunk 3]
```diff
@@ -275,42 +273,48 @@ - (CLLocationManager*) locationManager
...
+    
+    NSDate *currentTime = [NSDate date];
     self.aqiResponse = [[NSMutableDictionary alloc]init];
     
     __weak APHAirQualityDataModel *weakSelf = self;
     
-    if (!self.fetchingAirQualityReport && ((currentTime - lastAQICheckedTime) >= kAQICheckInterval)) {
+    if (!self.fetchingAirQualityReport && (([currentTime timeIntervalSinceDate: lastAQICheckedTime]) >= kAQICheckInterval)) {
+        [[NSUserDefaults standardUserDefaults]setObject:[NSDate new] forKey:kAQILastChecked];
+        [[NSUserDefaults standardUserDefaults]synchronize];
```
Comment (YuanZhu-apple) [inline Asthma/TabBarControllers/Dashboard/APHAirQualityDataModel.m]: Fix up spacing throughout this patch? add a space between "]" and "synchronize".

File: Asthma/Startup/APHAppDelegate.m [hunk 4]
```diff
@@ -261,8 +357,320 @@ - (NSArray *)offsetForTaskSchedules
...
+                }
+                
+            }
+            
+            // Get the difference in seconds between the start and end date for the sample
+            NSDateComponents* secondsSpentInBedOrAsleep = [[NSCalendar currentCalendar] components:NSCalendarUnitSecond
+                                                                                          fromDate:catSample.startDate
+                                                                                            toDate:catSample.endDate
+                                                                                           options:NSCalendarWrapComponents];
+            NSString*           quantityValue   = [NSString stringWithFormat:@"%ld", (long)secondsSpentInBedOrAsleep.second];
```
Comment (YuanZhu-apple) [inline Asthma/Startup/APHAppDelegate.m]: Same issue as https://github.com/ResearchKit/GlucoSuccess/pull/3/files#r30483440

File: Asthma.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -818,6 +819,22 @@
...
+			inputPaths = (
+			);
+			outputPaths = (
+			);
+			runOnlyForDeploymentPostprocessing = 0;
+			shellPath = /bin/sh;
+			shellScript = "${SRCROOT}/AppCore/APCAppCore/APCAppCore/Resources/get_debug_ip_address.sh";
+		};
+/* End PBXShellScriptBuildPhase section */
+
```
Comment (YuanZhu-apple) [inline Asthma.xcodeproj/project.pbxproj]: Public apps shouldn't have this build phase.

**Comment by ghost** [comment]
@YuanZhu-apple We have taken care of all comments on this PR.

File: Asthma/TabBarControllers/Dashboard/APHAirQualityDataModel.m [hunk 6]
```diff
@@ -275,42 +273,48 @@ - (CLLocationManager*) locationManager
...
-                    APCLogEventWithData(kNetworkEvent, (@{
-                                                          @"event_detail" : @"Received Air Quality Info from Server"
-                                                          }));
-                    
-                    //sends a new AQIAlert to the dashboard. The dashboard should receive it on the main thread.
-                    [weakSelf.airQualityReportReceiver airQualityModel:weakSelf didDeliverAirQualityAlert:weakSelf.aqiObject];
+        [self.networkManager get:kAlertGetJson headers:nil parameters:@{@"lat":@(lat),@"lon":@(lon)} completion:^(NSURLSessionDataTask __unused *task, id responseObject, NSError *error) {
+            APCLogError2(error);
+            weakSelf.fetchingAirQualityReport = NO;
+            if (!error) {
```
Comment (YuanZhu-apple) [inline Asthma/TabBarControllers/Dashboard/APHAirQualityDataModel.m:302]: In block use ‘strongSelf’ retains it, and it stays retained until the block returns.

```
// In block
__typeof__(self) strongSelf = weakSelf;
[strongSelf doSomething];
```

File: Asthma/TabBarControllers/Dashboard/APHAirQualityDataModel.m [hunk 7]
```diff
@@ -275,42 +273,48 @@ - (CLLocationManager*) locationManager
...
+                weakSelf.aqiResponse = [[NSMutableDictionary alloc]initWithDictionary:responseObject];
+                
+                //send notification for [APHDashboardViewController to prepareData];
+                APCLogEventWithData(kNetworkEvent, (@{
+                                                      @"event_detail" : @"Received Air Quality Info from Server"
+                                                      }));
+                
+                //sends a new AQIAlert to the dashboard. The dashboard should receive it on the main thread.
+                dispatch_async(dispatch_get_main_queue(), ^{
+                    if ([weakSelf.airQualityReportReceiver respondsToSelector:@selector(airQualityModel:didDeliverAirQualityAlert:)]) {
```
Comment (YuanZhu-apple) [inline Asthma/TabBarControllers/Dashboard/APHAirQualityDataModel.m]: In block use ‘strongSelf’.

File: Asthma/Startup/APHAppDelegate.m [hunk 8]
```diff
@@ -261,8 +302,318 @@ - (NSArray *)offsetForTaskSchedules
...
+        return stringToWrite;
+    };
+    
+    NSString*(^CategoryDataSerializer)(id) = ^NSString*(id dataSample)
+    {
+        HKCategorySample*   catSample       = (HKCategorySample *)dataSample;
+        NSString*           stringToWrite   = nil;
+        
+        if ([catSample.categoryType.identifier isEqualToString:@"HKCategoryTypeIdentifierSleepAnalysis"])
+        {
```
Comment (YuanZhu-apple) [inline Asthma/Startup/APHAppDelegate.m:468]: Should use the const value `HKCategoryTypeIdentifierSleepAnalysis` from HKTypeIdentifiers.h

File: Asthma/TabBarControllers/Dashboard/APHDashboardViewController.m [hunk 9]
```diff
@@ -150,17 +157,17 @@ - (void)prepareScoringObjects
...
         HKQuantityType *hkQuantity = [HKQuantityType quantityTypeForIdentifier:HKQuantityTypeIdentifierStepCount];
         self.stepScore = [[APCScoring alloc] initWithHealthKitQuantityType:hkQuantity
-                                                                        unit:[HKUnit countUnit]
-                                                                numberOfDays:-kNumberOfDaysToDisplay];
+                                                                      unit:[HKUnit countUnit]
+                                                              numberOfDays:-kNumberOfDaysToDisplay];
     }
     
     {
         self.peakScore = [[APCScoring alloc] initWithTask:@"DailyPrompt-27829fa5-d731-4372-ba30-a5859f655297"
```
Comment (YuanZhu-apple) [inline Asthma/TabBarControllers/Dashboard/APHDashboardViewController.m]: Use const value `kDailySurveyTaskID`

File: Asthma/Startup/APHAppDelegate.m [hunk 10]
```diff
@@ -200,7 +240,7 @@ - (void) setUpAppAppearance
                                                  @"YourAsthma-cc06cd68-b943-11e4-a71e-12e3f512a338" : [UIColor appTertiaryPurpleColor],
                                                  @"AsthmaHistory-d6d07ba4-b943-11e4-a71e-12e3f512a338" : [UIColor appTertiaryPurpleColor],
                                                  @"APHEnrollmentForRecontactTaskViewController-1E174065-5B02-11E4-8ED6-0800200C9A66" : [UIColor appTertiaryPurpleColor],
-                                                 @"AboutYou-27829fa5-d731-4372-ba30-a5859f688297" : [UIColor appTertiaryPurpleColor]
+                                                 @"AboutYou-27829fa5-d731-4372-ba30-a5859f688297" : [UIColor appTertiaryPurpleColor],
                                                  }];
```
Comment (YuanZhu-apple) [inline Asthma/Startup/APHAppDelegate.m:243]: Create constants for these identifiers? They are used more than once in this project.

**Comment by ghost** [comment]
@YuanZhu-apple We have taken care of your comments on this PR.

File: Asthma/Startup/APHAppDelegate.m [hunk 11]
```diff
@@ -261,8 +302,318 @@ - (NSArray *)offsetForTaskSchedules
...
+        NSDate*     consentDate = nil;
+        
+        if (user.consentSignatureDate)
+        {
+            consentDate = user.consentSignatureDate;
+        }
+        else
+        {
+            NSFileManager*  fileManager = [NSFileManager defaultManager];
+            NSString*       filePath    = [[self applicationDocumentsDirectory] stringByAppendingPathComponent:@"db.sqlite"];
```
Comment (YuanZhu-apple) [inline Asthma/Startup/APHAppDelegate.m]: Sorry, the last thing: `kDatabaseName`

**Pull request merged**